docs/sources/operators-guide/configuring/reference-configuration-parameters/index.md

```diff
@@ -1408,32 +1408,74 @@ query_frontend:
   # CLI flag: -ruler.query-frontend.address
   [address: <string> | default = ""]
 
-  # (advanced) Set to true if query-frontend connection requires TLS.
-  # CLI flag: -ruler.query-frontend.tls-enabled
-  [tls_enabled: <boolean> | default = false]
-
-  # (advanced) Path to the client certificate file, which will be used for
-  # authenticating with the server. Also requires the key path to be configured.
-  # CLI flag: -ruler.query-frontend.tls-cert-path
-  [tls_cert_path: <string> | default = ""]
-
-  # (advanced) Path to the key file for the client certificate. Also requires
-  # the client certificate to be configured.
-  # CLI flag: -ruler.query-frontend.tls-key-path
-  [tls_key_path: <string> | default = ""]
-
-  # (advanced) Path to the CA certificates file to validate server certificate
-  # against. If not set, the host's root CA certificates are used.
-  # CLI flag: -ruler.query-frontend.tls-ca-path
-  [tls_ca_path: <string> | default = ""]
-
-  # (advanced) Override the expected name on the server certificate.
-  # CLI flag: -ruler.query-frontend.tls-server-name
-  [tls_server_name: <string> | default = ""]
-
-  # (advanced) Skip validating server certificate.
-  # CLI flag: -ruler.query-frontend.tls-insecure-skip-verify
-  [tls_insecure_skip_verify: <boolean> | default = false]
+  grpc_client_config:
```

Comment:
We have a convention to name CLI flags and YAML config options consistently. Here you added a block config for GRPC client settings, but CLI flags have not been named accordingly. My suggestion is to rename CLI flags to have the prefix `-ruler.query-frontend.grpc-client-config.` instead of `-ruler.query-frontend.`.

That being said, I personally find the `_config` suffix useless (everything here is a config). Unfortunately in the past we already used `grpc_client_config`, but there's no strong consistency: in other places we just call it `client`. Between the two, I would more keen to name it `client` instead of `grpc_client_config` but I don't feel super strong about it.